File: HW4/userid/PekinDuck.cpp
#include <iostream>
#include "PekinDuck.h"
#include "Quack.h"
#include "FlysPoorly.h"
#include "LaysEggsNotBroody.h"

PekinDuck::PekinDuck( ) {
    flyBehavior = new FlysPoorly( );
    quackBehavior = new Quack( );
    eggBehavior = new LaysEggsNotBroody( );
}

void PekinDuck::display( ) {
    std::cout << "I'm a PekinDuck" << std::endl;
}

File: HW14/huan1588/PekinDuck.java
public class PekinDuck extends Duck {
    public PekinDuck( ) {
        quackBehavior = new Quack( );
        flyBehavior = new FlysPoorly( );
        eggBehavior = new LaysEggsNotBroody( );
    }

    public void display( ) {
        System.out.println("I'm a real pekin duck");
    }
}
